internal/handler/handler_test.go

```diff
@@ -230,3 +236,401 @@ func TestNewHtmlPdfService(t *testing.T) {
+			},
+			validateFunc: func(x *httptest.ResponseRecorder) {
+				if x.Code != http.StatusBadRequest {
+					t.Errorf("want %v got %v", http.StatusBadRequest, x.Code)
+				}
+				var r respModel.Response
+				got, err := io.ReadAll(x.Body)
+				json.Unmarshal(got, &r)
+				diff := testutil.Diff(r, respModel.Response{
+					Status:  http.StatusBadRequest,
+					Message: codes.GetErr(codes.ErrFileSizeExceeded),
+					Data:    nil,
+				})
+				if diff != "" {
+					t.Error(testutil.Callers(), diff)
+				}
+				diff = testutil.Diff(err, nil)
+				if diff != "" {
+					t.Error(testutil.Callers(), diff)
+				}
+			},
+		},
+		{
+			name: "Failure:: Upload:: no file found",
+			setupFunc: func() (*http.Request, *htmlPdfService) {
+				b := new(bytes.Buffer)
+				y := multipart.NewWriter(b)
+				part, _ := y.CreateFormFile("f", "some-file")
+				part.Write([]byte("abc"))
+				y.Close()
+				r := httptest.NewRequest(http.MethodPost, "/v1/register", b)
+				r.Header.Set("Content-Type", y.FormDataContentType())
+				mockLogicier := mock.NewMockHtmlPdfServiceLogicIer(mockCtrl)
+				rec := &htmlPdfService{
+					logic: mockLogicier,
+				}
+				return r, rec
+			},
+			validateFunc: func(x *httptest.ResponseRecorder) {
+				if x.Code != http.StatusBadRequest {
+					t.Errorf("want %v got %v", http.StatusBadRequest, x.Code)
+				}
+				var r respModel.Response
+				got, err := io.ReadAll(x.Body)
+				json.Unmarshal(got, &r)
+				diff := testutil.Diff(r, respModel.Response{
+					Status:  http.StatusBadRequest,
+					Message: codes.GetErr(codes.ErrFileParseFail),
+					Data:    nil,
+				})
+				if diff != "" {
+					t.Error(testutil.Callers(), diff)
+				}
+				diff = testutil.Diff(err, nil)
+				if diff != "" {
+					t.Error(testutil.Callers(), diff)
+				}
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			r, rec := tt.setupFunc()
+			w := httptest.NewRecorder()
+			x := rec.Upload
+			x(w, r)
+			tt.validateFunc(w)
+		})
+	}
+}
+
+func TestConvertToPdf(t *testing.T) {
+	mockCtrl := gomock.NewController(t)
+	defer mockCtrl.Finish()
+	tests := []struct {
+		name         string
+		requestBody  string
+		setupFunc    func() (*http.Request, *htmlPdfService)
+		validateFunc func(*httptest.ResponseRecorder)
+	}{
+		{
+			name:        "Success:: ConvertToPdf",
+			requestBody: "1",
+			setupFunc: func() (*http.Request, *htmlPdfService) {
+				var temp struct {
+					Values struct {
+						Name  string `json:"name"`
+						Marks int    `json:"marks"`
+						ID    string `json:"id"`
+					} `json:"values"`
+				}
+				temp.Values.Name = "vatsal"
+				temp.Values.Marks = 90
+				b, err := json.Marshal(temp)
+				if err != nil {
+				}
+				r := httptest.NewRequest(http.MethodPost, "/v1/generate/1", bytes.NewBuffer(b))
+				r = mux.SetURLVars(r, map[string]string{"id": "1"})
+				mockLogicier := mock.NewMockHtmlPdfServiceLogicIer(mockCtrl)
+				mockLogicier.EXPECT().HtmlToPdf(gomock.Any(), gomock.Any()).Times(1).
```

Comment:
The 2nd argument to `HtmlToPdf` is a concrete struct. This can be compared instead of using gomock.Any().

Reply:
done

Reply:
This is still pending....NOT DONE.

Reply:
it is done but there is a doubt

Reply:
done

Reply:
done